network/src/p2p/builder.rs

```diff
@@ -73,19 +77,34 @@ impl TransportBuilder {
     }
 
     /// Builds libp2p transport
-    pub fn build<TTransport>(
-        self,
-        transport: TTransport,
-    ) -> (transport::Boxed<(PeerId, StreamMuxerBox)>, PeerId)
-    where
-        TTransport: Transport + Sized + Clone + Send + Sync + 'static,
-        <TTransport as Transport>::Output: AsyncRead + AsyncWrite + Send + Unpin + 'static,
-        TTransport::Dial: Send + 'static,
-        TTransport::Listener: Send + 'static,
-        TTransport::ListenerUpgrade: Send + 'static,
-        TTransport::Error: Send + Sync,
-    {
-        (
+    pub fn build(self) -> Result<(BoxedP2PTransport, PeerId), std::io::Error> {
+        let transport = {
+            cfg_if::cfg_if! {
+                if #[cfg(target_arch = "wasm32")] {
+                    use libp2p::wasm_ext;
+
+                    // Note that DNS has been implictly supported in the extended javascript code,
+                    // and TCP is not feasible in browsers
+                    wasm_ext::ExtTransport::new(wasm_ext::ffi::websocket_transport())
+                } else {
+                    // Choose tokio over async-std here for 2 reasons:
+                    //  1. Tokio has better performance as an coroutine scehduler.
+                    //  2. TokioDnsConfig does not propagate async to build function's signature, while DnsConfig does.
+                    // Cons:
+                    //  1. Tokio builds more slowly.
+                    //  2. Tokio's API is slightly more complicated.
+                    //
```

Comment:
I'm not sure if the comment belongs here. We are fully switching to tokio right and dropping async-std? I think this is a great call but probably a discussion for the PR comments rather than code comments

Reply:
Good to have a discussion, I will post in the channel